dipy/direction/closest_peak_direction_getter.pyx

```diff
@@ -58,6 +57,73 @@ cdef int closest_peak(np.ndarray[np.float_t, ndim=2] peak_dirs,
             return 0
     return 1
 
+cdef extern from "dpy_math.h" nogil:
+    int dpy_signbit(double x)
+    double dpy_rint(double x)
+    double fabs(double)
+
+@cython.cdivision(True)
+cdef inline double stepsize(double point, double increment) nogil:
+    """Compute the step size to the closest boundary in units of increment."""
+    cdef:
+        double dist
+    dist = dpy_rint(point) + .5 - dpy_signbit(increment) - point
+    if dist == 0:
+        # Point is on an edge, return step size to next edge.  This is most
+        # likely to come up if overstep is set to 0.
+        return 1. / fabs(increment)
+    else:
+        return dist / increment
+
+cdef void step_to_boundary(double * point, double * direction,
+                           double overstep) nogil:
+    """Takes a step from point in along direction just past a voxel boundary.
+
+    Parameters
+    ----------
+    direction : c-pointer to double[3]
+        The direction along which the step should be taken.
+    point : c-pointer to double[3]
+        The tracking point which will be updated by this function.
+    overstep : double
+        It's often useful to have the points of a streamline lie inside of a
+        voxel instead of having them lie on the boundary. For this reason,
+        each step will overshoot the boundary by ``overstep * direction``.
+        This should not be negative.
+
+    """
+    cdef:
+        double step_sizes[3]
+        double smallest_step
+
+    for i in range(3):
+        step_sizes[i] = stepsize(point[i], direction[i])
+
+    smallest_step = step_sizes[0]
+    for i in range(1, 3):
+        if step_sizes[i] < smallest_step:
+            smallest_step = step_sizes[i]
+
+    smallest_step += overstep
+    for i in range(3):
+        point[i] += smallest_step * direction[i]
+
+cdef void fixed_step(double * point, double * direction, double step_size) nogil:
+    """Updates point by stepping in direction.
+
+    Parameters
+    ----------
+    direction : c-pointer to double[3]
+        The direction along which the step should be taken.
+    point : c-pointer to double[3]
+        The tracking point which will be updated by this function.
+    step_size : double
+        The size of step in units of direction.
+
+    """
+    for i in range(3):
+        point[i] += direction[i] * step_size
+
```

Comment:
I do not understand why you need this block in `close_peak_direction_getter.pyx`. it is not used.

Reply:
we don't need it. This should have been removed when we moved `generate_streamline(.)` to the parent class `DirectionGetter`. thx.

Reply:
Ok, Great!  This PR is ready for me. 

There are 2-3 things that can be improved (unnecessary copy) but we can do that in a new PR.